runtime/compiler/z/runtime/PicBuilder.m4

```diff
@@ -1702,6 +1702,48 @@ LABEL(L_virtualDispatchExit)
 
     END_FUNC(_virtualUnresolvedHelper,virUH,7)
 
+ZZ ================================================
+ZZ  PICBuilderRoutine : _jitResolveConstantDynamic
+ZZ ================================================
+    START_FUNC(_jitResolveConstantDynamic,jRCD)
+
+LABEL(_jitResolveConstantDynamicBody)
+ZZ Loading arguments for JIT helper
+ZZ R1 - Address of Constant Pool
+ZZ R2 - CPIndex
+ZZ R3 - JIT'd code Return Address
+    L_GPR r1,eq_cp_inDataSnippet(,r14)
+    LGF_GPR r2,eq_cpindex_inDataSnippet(,r14)
+    L_GPR r3,eq_codeRA_inDataSnippet(,r14)
+
+ZZ Following snippet prepares JIT helper call sequence
+ZZ which needs r14 to be free which holds address of
+ZZ Snippet from where this is called. 
+ZZ As jitResolveConstantDynamic is called via 
+ZZ SLOW_PATH_ONLY_HELPER glue, it is guranteed that
+ZZ all the registers (volatile and non-volatile) are preserved.
+ZZ This allows us to use R0 to preserve R14 
+LOAD_ADDR_FROM_TOC(rEP,TR_S390jitResolveConstantDynamic)
+    LR_GPR r0,r14
+    BASR r14,rEP
+    LR_GPR r14,r0
+
+ZZ Load the address of literal pool
+ZZ Store the return value of helper call which
+ZZ is address of resolved constant dynamic 
+ZZ into the Literal Pool 
+    L_GPR   r1,eq_literalPoolAddr_inDataSnippet(r14)
+    ST_GPR  r2,0(,r1)
+
+    L_GPR r2,eq_codeRA_inDataSnippet(,r14) # Get mainline RA
+
+ZZ Branch instruction in mainline will be patched here with NOP
+    LHI r1,HEX(04)
+    STCY r1,-5(r2)
+
+    L_GPR r14,eq_codeRA_inDataSnippet(,r14) 
```

Comment:
This load seems a bit redundant to the one above at 1738.  Perhaps we can consolidate the two?

Reply:
We can generate that sequence. 
It will look like,
```
L_GPR r14,eq_codeRA_inDataSnippet(,r14)
LHI r1,HEX(04)
STCY r1,-5(r14)
BR r14
```

Reply:
Yeah, that looks good.

Reply:
Might as well do it to shorten the path length and improve readibility given we're waiting for the OMR change to propagate anyway.